mycore-base/src/main/java/org/mycore/resource/provider/MCRResourceProvider.java

```diff
@@ -102,30 +110,57 @@ public BaseDirPrefixStripper(File baseDir) {
         }
 
         @Override
-        public String doStrip(String value) {
+        public List<String> getStrippedPaths(String value) {
             if (value.startsWith(prefix)) {
-                return value.substring(prefix.length());
+                return List.of(value.substring(prefix.length()));
             }
-            return null;
+            return Collections.emptyList();
         }
 
         @Override
         public String toString() {
             return prefix;
         }
 
-        public static Set<PrefixStripper> ofClassLoader(ClassLoader classLoader) {
-            Set<PrefixStripper> strippers = new LinkedHashSet<>();
-            List<URI> classpath = new ClassGraph().overrideClassLoaders(classLoader).getClasspathURIs();
-            classpath.forEach(uri -> {
-                if (uri.getScheme().equals("file")) {
-                    File file = new File(uri.getPath());
-                    if (file.isDirectory()) {
-                        strippers.add(new BaseDirPrefixStripper(file));
+    }
+
+    final class ClassLoaderPrefixStripper extends PrefixStripperBase {
+
+        private static final int CLASS_GRAPH_THREADS = 4;
```

Comment:
Why 4? Shouldn't that be configurable?

Reply:
Good point, wanted to add a configuration value and then forgot. The ClassGraph library has a nifty formula to calculate `ClassGraph#DEFAULT_NUM_WORKER_THREADS`, but it can not be reused, because that field is package-private.